lib/python/picongpu/picmi/diagnostics/openpmd_sources/charge_density.py

```diff
@@ -0,0 +1,59 @@
+"""
+This file is part of PIConGPU.
+Copyright 2021-2025 PIConGPU contributors
+Authors: Masoud Afshari
+License: GPLv3+
+"""
+
+from .... import util
+from .source_base import SourceBase
+from ...pypicongpu.output.openpmd_source import ChargeDensity as PyPIConGPUChargeDensity
+import typeguard
+import typing
+
+
+@typeguard.typechecked
+class ChargeDensity(SourceBase):
+    """
+    Represents the charge density data source for openPMD output in particle-in-cell simulations.
+
+    This class defines the charge density field, derived from particle species at runtime,
+    which can be output using the openPMD standard. An optional filter can be applied to select
+    which particles contribute to the charge density.
+
+    Parameters
+    ----------
+    filter: str, optional
+        Name of a filter to select particles contributing to the charge density.
+        Default: None (all valid particles).
+    """
+
+    filter = util.build_typesafe_property(typing.Optional[str])
+
+    def __init__(self, filter: typing.Optional[str] = None):
+        self.filter = filter
+        self.check()
```

Comment:
default filter should be `all`, as this is the PIConGPU standard. And since we will stick to str for now, I would like to avoid using Optional here as it only complicates the implementation.

In addition, this class is missing the option to configure the species contributing to this source, please add a list of species that should contribute to this source.